swap/src/bitcoin.rs

```diff
@@ -1,202 +1,288 @@
-use anyhow::{Context, Result};
+use anyhow::{anyhow, bail, Result};
 use async_trait::async_trait;
-use backoff::{backoff::Constant as ConstantBackoff, future::FutureOperation as _};
-use bitcoin::util::psbt::PartiallySignedTransaction;
-use bitcoin_harness::{bitcoind_rpc::PsbtBase64, BitcoindRpcApi};
-use reqwest::Url;
-use std::time::Duration;
-use tokio::time::interval;
-use xmr_btc::{
-    bitcoin::{
-        BroadcastSignedTransaction, BuildTxLockPsbt, GetBlockHeight, SignTxLock,
-        TransactionBlockHeight, WatchForRawTransaction,
-    },
-    config::Config,
-};
-
-pub use ::bitcoin::{Address, Transaction};
-pub use xmr_btc::bitcoin::*;
-
-pub const TX_LOCK_MINE_TIMEOUT: u64 = 3600;
-
-#[derive(Debug)]
-pub struct Wallet {
-    pub inner: bitcoin_harness::Wallet,
-    pub network: bitcoin::Network,
-}
-
-impl Wallet {
-    pub async fn new(name: &str, url: Url, network: bitcoin::Network) -> Result<Self> {
-        let wallet = bitcoin_harness::Wallet::new(name, url).await?;
-
-        Ok(Self {
-            inner: wallet,
-            network,
-        })
+use bitcoin::hashes::{hex::ToHex, Hash};
+use ecdsa_fun::{adaptor::Adaptor, fun::Point, nonce::Deterministic, ECDSA};
+use miniscript::{Descriptor, Segwitv0};
+use rand::{CryptoRng, RngCore};
+use serde::{Deserialize, Serialize};
+use sha2::Sha256;
+use std::str::FromStr;
+
+use crate::{config::Config, ExpiredTimelocks};
+
+use crate::bitcoin::timelocks::{BlockHeight, Timelock};
```

Comment:
```suggestion
use std::str::FromStr;
use crate::{config::Config, ExpiredTimelocks};
use crate::bitcoin::timelocks::{BlockHeight, Timelock};
```

Reply:
hi, I don't understand this suggestion due to the code formatting. Do you want the `use crate::*` imports to NOT be separated from the `use *` imports with a line?

Reply:
I don't mind the segregation of the `use crate::*`.

Do note that in the original code there is a newline between both `use crate::*`, that needs to be removed.